mlflow/tracing/export/mlflow.py

```diff
@@ -0,0 +1,146 @@
+import json
+import logging
+import threading
+from contextvars import ContextVar
+from typing import Any, Dict, Optional, Sequence
+
+from opentelemetry.sdk.trace.export import SpanExporter
+
+from mlflow.tracing.client import TraceClient
+from mlflow.tracing.types.model import Span, Trace, TraceData, TraceInfo
+from mlflow.tracing.types.wrapper import MLflowSpanWrapper
+
+_logger = logging.getLogger(__name__)
+
+
+class MLflowSpanExporter(SpanExporter):
+    """
+    An exporter implementation that logs the traces to MLflow.
+
+    MLflow backend (will) only support logging the complete trace, not incremental updates
+    for spans, so this exporter is designed to aggregate the spans into traces in memory.
+    Therefore, this only works within a single process application and not intended to work
+    in a distributed environment. For the same reason, this exporter should only be used with
+    SimpleSpanProcessor.
+
+    If we want to support distributed tracing, we should first implement an incremental trace
+    logging in MLflow backend, then we can get rid of the in-memory trace aggregation.
+    """
+
+    _MAX_CHARS_IN_TRACE_INFO_INPUTS_OUTPUTS = 300  # TBD
+
+    def __init__(self, client: TraceClient):
+        self._client = client
+        self._trace_aggregator = InMemoryTraceDataAggregator.get_instance()
+
+    def export(self, spans: Sequence[MLflowSpanWrapper]):
+        """
+        Export the spans to MLflow backend.
+
+        Args:
+            spans: A sequence of MLflowSpanWrapper objects to be exported. The base
+                OpenTelemetry (OTel) exporter should take OTel spans but this exporter
+                takes the wrapper object, so we can carry additional MLflow-specific
+                information such as inputs and outputs.
+        """
+        mlflow_spans = []
+        for span in spans:
+            if not isinstance(span, MLflowSpanWrapper):
+                _logger.warning(
+                    "Span exporter expected MLflowSpanWrapper, but got "
+                    f"{type(span)}. Skipping the span."
+                )
+            mlflow_span = span._to_mlflow_span()
+            self._trace_aggregator.add_span(mlflow_span)
+            mlflow_spans.append(mlflow_span)
+
+        # Call this after processing all spans because the parent-child order might
+        # not be preserved in the input spans
+        for span in mlflow_spans:
+            if span.parent_span_id is None:
+                self._export_trace(span)
```

Comment:
The order seems don't matter as here you're just checking the parent_span_id field of a span, if it's None then export it.
Q: just want to make sure that there's no change we get multiple root spans? Maybe an exception catch would be helpful, just a little bit concerned

Reply:
> The order seems don't matte

The order here means the two steps of (1) add all spans -> (2) export root, instead  of just doing them in one loop. Basically we want to make sure doing export after all spans are added to trace.

> Q: just want to make sure that there's no change we get multiple root spans?

Multiple root spans are totally valid, they should be exported as two separate traces.

Reply:
Yea I agree, while the comment is a little bit misleading about the `parent-child order` part.

> Multiple root spans are totally valid, they should be exported as two separate traces.

Ah yes! I was about to say multiple root spans with the same trace_id, then I realized we're the one who creates trace, Otel just has span concept, so no questions now :)

Reply:
Will rephrase the comment then!